nextcord/command_client.py

```diff
@@ -0,0 +1,396 @@
+
+    async def invoke(self, interaction: Interaction, **kwargs):
+        # Invokes the callback with the kwargs given.
+        await self._callback(interaction, **kwargs)
+
+    @property
+    def payload(self) -> dict:
+        ret = dict()
+        ret["type"] = self.type.value
+        ret["name"] = self.name
+        ret["description"] = self.description
+        if self.required is not None:
+            ret["required"] = self.required
+        if self.choices:
+            ret["choices"] = [{key: value} for key, value in self.choices.items()]
+        if self.children:
+            ret["options"] = [child.payload for child in self.children.values()]
+        elif self.arguments:
+            ret["options"] = [argument.payload for argument in self.arguments.values()]
+        return ret
+
+    def subcommand(self, **kwargs):
+        def decorator(func: Callable):
+            result = ApplicationSubcommand(func, self, CommandOptionType.SUB_COMMAND, **kwargs)
+            self.children[result.name] = result
+            return result
+        return decorator
+
+    def group(self, **kwargs):
+        warn("This function will be removed for application commands.", DeprecationWarning, stacklevel=2)
+        return self.subcommand(**kwargs)
+
+
+class ApplicationCommand(ApplicationSubcommand):
+    def __init__(self, callback: Callable, cmd_type: CommandType,
+                 name: str = "", description: str = "", guild_ids: List[int] = None,
+                 default_permission: Optional[bool] = None):
+        super().__init__(callback=callback, parent=None, cmd_type=cmd_type, name=name, description=description,
+                         guild_ids=None)
+        self._state: Optional[ConnectionState] = None
+        if guild_ids is None:
+            guild_ids = list()
+        if not asyncio.iscoroutinefunction(callback):
+            raise TypeError("Callback must be a coroutine.")
+
+        self.default_permission: Optional[bool] = default_permission
+        self.guild_ids: List[int] = guild_ids
+        self.type = cmd_type
+        self.id: Optional[int] = None
+
+    def parse_response(self, response: ApplicationCommandResponse):
+        print(self.id)
+        # TODO: Grab state and have access to get_guild stuff for get_member and get_message for user/message commands.
+        self.id = response.id
+        self._state = response._state
+
+    @property
+    def payload(self) -> Union[List[Dict[str, ...]], Dict[str, ...]]:
+        ret = super().payload
+        if self.default_permission is not None:
+            ret["default_permission"] = self.default_permission
+
+        if self.guild_ids:
+            guild_ret = list()
+            for guild_id in self.guild_ids:
+                temp = ret.copy()
+                temp["guild_id"] = guild_id
+                guild_ret.append(temp)
+            return guild_ret
+        else:
+            return ret
+
+
+class CommandClient(Client):
```

Comment:
I'm not sure this is a good idea...

Making another Client class to do slash commands is the one thing everyone hated Pycord for.

Reply:
I don’t know exactly but people don't like 3rd party libs because of that, however as a fork we have control over the code unlike a 3rdplib doesn't